src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1461 @@
+*/
+void test_l3_HdmiIn_get_edid2allmsupport(void)
+{
+    gTestID = 17;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_MAX;
+    int32_t select;
+    bool allmsupport = false;
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+
+    printf(" \n================Please Select Inputs==================\n");
+
+    printf("\n*******Enter the port number to select*******\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select <= 0 || select > numInputPorts)
+    {
+       printf("\nInvalid port selected\n");
+       goto exit;
+    }
+
+    port = select;
+
+    ret = dsGetEdid2AllmSupport(port, &allmsupport);
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed dsSetEdid2AllmSupport Error:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    } else {
+           UT_LOG_INFO("Result : dsSetEdid2AllmSupport port type: %s, portnumber: %s\t"
+                         "retrieved allmsupport: %s",
+                 ut_control_GetMapString(dsHdmiInPort_mapTable, port), port,
+                 ut_control_GetMapString(bool_mapTable, allmsupport));
+    }
+
+exit:
+   UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+
+/**
+* @brief This test terminates the dsHdmiIn.
+*
+* This test function terminates the dsHdmiIn.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 018@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_dsHdmiIn_terminate(void)
+{
+    gTestID = 18;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsHdmiInTerm()");
+    ret = dsHdmiInTerm();
+    if(ret != dsERR_NONE) {
+        UT_LOG_ERROR("Failed dsHdmiInTerm() dsError_t:[%s]", ut_control_GetMapString(dsError_mapTable, ret));
+    }
+    else {
+        UT_LOG_INFO("Result dsHdmiInTerm() dsError_t:[%s]", ut_control_GetMapString(dsError_mapTable, ret));
+    }
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main test(s) for this module
+ *
+ * @return int - 0 on success, otherwise failure
+ */
+int test_l3_dsHdmiIn_register ( void )
+{
+    ut_kvp_status_t status = UT_KVP_STATUS_SUCCESS;
+    pSuite = UT_add_suite( "[L3 dsHdmiIn --  advanced]", NULL, NULL );
+
+    status = ut_kvp_getStringField(ut_kvp_profile_getInstance(), "dsHdmiIn.Type", gDeviceType, TEST_DS_DEVICE_TYPE_SIZE);
+
+    if (status == UT_KVP_STATUS_SUCCESS ) {
+        if (!strncmp(gDeviceType, TEST_TYPE_SOURCE_VALUE, TEST_DS_DEVICE_TYPE_SIZE)) {
+            // Create the test suite for source type
+            pSuite = UT_add_suite("[L3 dsHdmiIn - Source]", NULL, NULL);
+            if (pSuite == NULL)
```

Comment:
just use `assert( pSuite != NULL );` and make the code simple, it will never fail, and there's no advanctage of returning -1

Reply:
done